src/dscord/util/json.d

```diff
@@ -3,7 +3,195 @@
 */
 module dscord.util.json;
 
-public import fast.json : FastJson = Json, parseTrustedJSON, DataType;
+import std.traits;
+
 public import vibe.data.json : VibeJSON = Json, parseJsonString;
 
-alias JSON = FastJson!(0u, false);
+version (Have_fast) {
+  public import fast.json : FastJson = Json, parseTrustedJSON, DataType;
+
+  alias JSON = FastJson!(0u, false);
+
+  class JSONDecoder {
```

Comment:
**NB** this is not a interface w/ subclasses because I hit some idiot roadblock with the way interfaces link in D. For the purpose of not murdering myself debugging it, I decided to just implement it using a version block and then revisit this in the future.